Update to latest fedora-bootstrap (bootstrap5)

ryanlerch opened the pull request.

ryanlerch commented on mirrormanager2/app.py:
```diff
@@ -105,7 +105,7 @@ def create_app(config=None):
     # Flask-Admin does not support having a single instance and multiple calls
     # to init_app() (it stores the app as an instance attribute)
-    ADMIN = Admin(template_mode="bootstrap3")
+    ADMIN = Admin(template_mode="bootstrap4")
```
flask-admin doesnt have a bootstrap5 template mode, so we use the bootstrap4 one... it works ok enough for the admin pages

ryanlerch commented on mirrormanager2/templates/fedora/admin/master.html:
```diff
@@ -1,94 +1,24 @@
 {% extends 'admin/base.html' %}
 {% import 'admin/lib.html' as lib with context %}
 {% import 'admin/static.html' as admin_static with context %}
+{% from "_macros.html" import footer, headincludes, nav, flashmessages, bottomincludes %}
```
this template is why i used macros here -- cos the admin plugin basically needs its own master template, so used macros for the complex stuff here so we can re-use in the main master template

abompard commented on mirrormanager2/templates/fedora/host.html:
```diff
@@ -1,267 +1,246 @@
-      {{ asn.asn }}
+    {% if not host.private %}
+    <div><strong>Last Crawled:</strong> {% if host.last_crawled %} {{ host.last_crawled
+      }} <a href="{{url_for('base.crawler_log', host_id=host.id)}}">[Log]</a> {% else %} - {% endif %} </div>
```
Maybe "never" rather than "-" here?

ryanlerch replied:
yeah -- it was always '-' but never is better. fixed.

abompard reviewed (changes requested):
Looks great! Some remarks:
- all the flash messages for errors display in green, they should have the `error` category.
- you can drop the views and templates you no longer use, such as `host_category`.

ryanlerch commented on mirrormanager2/templates/fedora/export_compliance.html:
```diff
@@ -0,0 +1,26 @@
+<div class="alert alert-light form-text">
```
whoops -- yeah didnt mean to add that file -- just going to keep this macro in _macros.html

removed this file

ryanlerch commented on mirrormanager2/templates/fedora/maps_interactive.html:
```diff
@@ -1,4 +1,4 @@
-{% extends "master_noauth.html" %}
+{% extends "master" %}
```
thanks for the catch! fixed.

ryanlerch commented:
> Looks great! Some remarks:
> 
> * all the flash messages for errors display in green, they should have the `error` category.
added a bunch of error categories to the flask.flash() calls
> * you can drop the views and templates you no longer use, such as `host_category`.
dropped the host_category page!

ryanlerch commented:
Ready for re-review if you want

abompard commented:
Looks great, anything you want to change before we merge it?

abompard commented on mirrormanager2/templates/fedora/site.html:
```diff
@@ -1,120 +1,90 @@
+<div class="mt-3 d-flex justify-content-between align-items-center">
+  <div>
+    <h2 class="mb-0"><span class="fa fa-globe"></span> {{ site.name }}</h2>
+  Created by <strong>{{ site.created_by }}</strong> at <strong>{{ site.created_at }}</strong>
```
This looks a little bit weird:
![image](https://github.com/fedora-infra/mirrormanager2/assets/95780/e85224e8-9856-4c88-8616-b826d8f8a4ab)
Maybe do some formatting with the date? Nothing too fancy of course.

ryanlerch replied:
Yeah -- i started that, but involved adding a filter -- so will merge this, and fix in another PR
